### DISCUSSION TITLE
Form: How to Create a Static Choice

### DISCUSSION THREAD

**Discussion by nofue** [sqlpage/SQLPage#606] [Q&A] [answered]
My form has a field which stores the values -1, 0 and 1 to reflect three states. It would be nice to have a drop down list showing the labels of those states to help the user to decide. I guess this must be somewhere in the **'select' as type** documentation, but I can't find it ;) All I could find is creating an option list from querying another table, but this might be overkill in this case …

**Comment by lovasoa** [comment, marked as answer]
Hello! Isn't this example from the documentation what you need? 


```sql
select 
    'form'                        as component,
    'examples/show_variables.sql' as action;
select 
    'Fruit'  as name,
    'select' as type,
    TRUE     as searchable,
    1        as value,
    '[{"label": "Orange", "value": 0}, {"label": "Apple", "value": 1}, {"label": "Banana", "value": 3}]' as options;
```

https://sql.datapage.app/component.sql?component=form

**Reply by nofue** [reply]
Oops. Never found this, although I worked through two "fruit" examples. Thanks for your fast response.

**Comment by arcanosam** [comment]
> Hello! Isn't this example from the documentation what you need?
> 
> ```sql
> select 
>     'form'                        as component,
>     'examples/show_variables.sql' as action;
> select 
>     'Fruit'  as name,
>     'select' as type,
>     TRUE     as searchable,
>     1        as value,
>     '[{"label": "Orange", "value": 0}, {"label": "Apple", "value": 1}, {"label": "Banana", "value": 3}]' as options;
> ```
> 
> https://sql.datapage.app/component.sql?component=form

Very cool. Nothing about the topic, but is it possible to build that JSON dynamically?

**Reply by nofue** [reply]
I wonder why you might want to do that? 

The whole point of this kind of attribute is to create "segments" of data, in the case of the example fruits. If you have lots of possible segments, you better create a table where you put these items, looking them up from the table you want to be "segmented". If you then need more values to select from, you simply put them into the lookup table -- that's the way RDMBS should be dealt with.

What do you want to accomplish?

**Comment by arcanosam** [comment]
Just for curiosity to know if the SQLPage would understand such thing, an SQL statement instead of a string as like in the example.

**Reply by nofue** [reply]
I hope I got you right: So you don't want to create a JSON statement for just a limited set of options programmatically, right?

SQLPage always deals with SQL statements, JSON being the exception to the rule. The code to get the (filtered) content of a lookup table can be created with  **SELECT statement AS dropdown;** 

There is an example of this in the discussions, I think this is what you're asking for:
https://github.com/lovasoa/SQLpage/discussions/480
One of the comments will point you to the row-level parameter of form, 'downloads'. 
I hope this works for you.

**Comment by arcanosam** [comment]
Thanks for enlightenment. that discussion you linked solve my question.
